server/src/test/java/org/opensearch/transport/TransportLoggerTests.java

```diff
@@ -93,17 +94,14 @@ public void testLoggingHandler() throws Exception {
     private BytesReference buildRequest() throws IOException {
         boolean compress = randomBoolean();
         try (BytesStreamOutput bytesStreamOutput = new BytesStreamOutput()) {
-            OutboundMessage.Request request = new OutboundMessage.Request(
+            return NativeOutboundMessageTests.serializeNativeOutboundRequest(
```

Comment:
Could we also follow the same approach as `InboundDecoderTests` with `abstract serialize` method?

Reply:
I think its not needed for `TransportLoggerTests` since serialization is not the part that it is testing. The implementation being tested is for `logInboundMessage` and `logOutboundMessage` which just take in `BytesReference`. The native request is just being used to create bytes. WDYT?

Reply:
The idea behind `serialize` (or alternative) is to abstract over the protocol, how the `TransportLoggerTests` would deal with protobuf message? (since it directly uses `NativeOutboundMessageTests` here). To your point, `serialize` is really not a good fir here, but making this method `private BytesReference buildRequest() throws IOException` would be much better option.

Reply:
Makes sense!